core/src/main/java/com/amplifyframework/core/model/query/predicate/QueryPredicateGroup.java

```diff
@@ -103,6 +103,38 @@ public static QueryPredicateGroup not(QueryPredicateGroup predicate) {
         return new QueryPredicateGroup(Type.NOT, Arrays.asList(predicate));
     }
 
+    /**
+     * Evaluate the combination of operations associated with
+     * this group of predicates.
+     * @param object The object to evaluate against
+     * @return Evaluated result of this logical combination
+     * @throws IllegalArgumentException when the object contains
+     *          a field with data type that cannot be evaluated
+     */
+    @Override
+    public boolean evaluate(Object object) throws IllegalArgumentException {
+        switch (type) {
+            case OR:
+                for (QueryPredicate predicate : predicates) {
+                    if (predicate.evaluate(object)) {
+                        return true;
+                    }
+                }
+                return false;
+            case AND:
+                for (QueryPredicate predicate : predicates) {
+                    if (!predicate.evaluate(object)) {
+                        return false;
+                    }
+                }
+                return true;
+            case NOT:
+                return !predicates.get(0).evaluate(object);
```

Comment:
If predicates is empty, would cause a crash. Would be better to guard with a check.

Reply:
good point